drools-persistence/drools-persistence-jpa/src/test/java/org/drools/persistence/session/JpaPersistentStatefulSessionTest.java

```
@@ -99,521 +99,459 @@ public void tearDown() throws Exception {
+                " list.add( $i );\n" +
+                "end\n" +
+                "\n";
 
-        StatefulKnowledgeSession ksession = JPAKnowledgeService.newStatefulKnowledgeSession( kbase, null, env );
-        List<?> list = new ArrayList<Object>();
+        final KieBase kbase = new KieHelper().addContent(str, ResourceType.DRL).build();
 
-        ksession.setGlobal( "list",
-                            list );
+        KieSession ksession = KieServices.get().getStoreServices().newKieSession(kbase, null, env);
+        List<?> list = new ArrayList<>();
 
-        AtomicInteger value = new AtomicInteger(4);
-        FactHandle atomicFH = ksession.insert( value );
+        ksession.setGlobal("list", list);
+
+        final AtomicInteger value = new AtomicInteger(4);
+        FactHandle atomicFH = ksession.insert(value);
 
         ksession.fireAllRules();
 
-        assertEquals( 1,
-                      list.size() );
+        assertEquals(1, list.size());
 
         value.addAndGet(1);
         ksession.update(atomicFH, value);
         ksession.fireAllRules();
-        
-        assertEquals( 2,
-                list.size() );
-        String externalForm = atomicFH.toExternalForm();
-        
-        ksession = JPAKnowledgeService.loadStatefulKnowledgeSession(ksession.getIdentifier(), kbase, null, env);
-        
+
+        assertEquals(2, list.size());
+        final String externalForm = atomicFH.toExternalForm();
+
+        ksession = KieServices.get().getStoreServices().loadKieSession(ksession.getIdentifier(), kbase, null, env);
+
         atomicFH = ksession.execute(CommandFactory.fromExternalFactHandleCommand(externalForm));
-        
+
         value.addAndGet(1);
         ksession.update(atomicFH, value);
-        
+
         ksession.fireAllRules();
-        
+
         list = (List<?>) ksession.getGlobal("list");
-        
-        assertEquals( 3,
-                list.size() );
-        
+
+        assertEquals(3, list.size());
```

`AssertJ` is your friend. `Assertions.assertThat(list).hasSize(3);`
Let the test fail and see for yourself the vast difference in the failure message if you do it like that.